### ISSUE TITLE
AMP Optimizer transformHtml Returning Undefined Without Errors In Specific Node v14.16.0 ENV

### ISSUE THREAD

**Issue by SkylerRogers** [ampproject/amp-toolbox#1334]
Logic to reproduce on latest version (`"@ampproject/toolbox-optimizer": "2.8.0"`):
   
    const AmpOptimizer = require('@ampproject/toolbox-optimizer');
    
    const ampOptimizer = AmpOptimizer.create({
     preloadHeroImage: false,
     lts: false
    });

    const body = `<!DOCTYPE html>
      <html lang="en" ⚡>
        <head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1" />
          <title>Test</title>
        </head>
        <body id="top">
          <h2>Test</h2>
        </body>
      </html>`;

    const optimizedHtml = await ampOptimizer.transformHtml(body);
    
    optimizedHtml // returns undefined (no error output)
    

**Note:** The above works on `"@ampproject/toolbox-optimizer": "2.6.0"`

**Comment by LKWLaLa** [comment]
+1

**Comment by kaufmann42** [comment]
+1

**Comment by SkylerRogers** [comment]
**Another Note:** I tried this with no arguments (just `AmpOptimizer.create()`) per [docs](https://github.com/ampproject/amp-toolbox/tree/main/packages/optimizer#api) and got same behavior.

**Comment by sebastianbenz** [comment]
Thanks for reporting. Have you tried updating to the latest version 2.9.0?

**Comment by SkylerRogers** [comment]
@sebastianbenz - apologies here. I have confirmed the library works in isolation. It's not working in a jest/test-library environment within our application. Trying to replicate with most minimal setup possible. Will keep you posted on findings.

**Comment by sebastianbenz** [comment]
😅 thanks for the update

**Comment by SkylerRogers** [comment]
Another update: Was able to replicate issue on our side, outside of @testing-library/Jest. However, was able to isolate within @testing-library/Jest and the library worked as expected. 

TLDR: I think the surface area of this issue is getting smaller and smaller and specific to our setup. Will keep you posted as I find more info.

**Comment by SkylerRogers** [comment]
Another small update: if I do an NPM link of the latest amp-toolbox clone against our repository, things work as expected on our side. However, if I add any log statements/logic into the linked toolbox clone, I am not seeing them trigger. Working now to see what could be happening. 

Is there anything we need to do outside of regular `npm link` in toolbox and `npm link @amp-toolbox/optimizer` on our side to get your lib to log any custom local code?

**Comment by SkylerRogers** [comment]
@sebastianbenz I think we got it figured out via nuking our third party artifactory cache & deleting/regenerating `npm-shrinkwrap.json / package-lock.json`. Sorry for all the back and forth here. Thanks again for the help here 🙏
